File: tests/fixtures/compare/sample.ts
// Realistic TS code with various issues
const data: any = [1, 2, 3];
var count = 0;

function process(item: any): void {
  if (item == null) return;
  console.log("processing", item);
  count++;
}

function empty(): void {}

function unreachable(): number {
  return 42;
  const x = 1;
}

debugger;

if (true) {
  eval("alert(1)");
}

if (count === NaN) {}

File: tests/fixtures/compare/sample.py
# Realistic Python code with various issues for linter comparison
from os import *
import sys

DEBUG = True

def process(items, cache={}):
    global DEBUG
    for item in items:
        print(f"processing {item}")
        if item > 100:
            if item > 200:
                pass

    try:
        result = sum(items)
    except:
        pass

    try:
        x = 1 / 0
    except ValueError:
        raise

    eval("print('hello')")

    assert len(items) > 0

    return result

class Broken:
    def __init__(self):
        return 42
